owncloudApp/src/main/java/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.kt

```diff
@@ -272,35 +262,190 @@ class ErrorMessageAdapter {
+                RemoteOperationResult.ResultCode.SHARE_NOT_FOUND -> {
+                    if (operationType == OperationType.CREATE_PUBLIC_SHARE)
+                        return f.format(R.string.share_link_file_no_exist)
+                    if (operationType == OperationType.REMOVE_SHARE)
+                        return f.format(R.string.unshare_link_file_no_exist)
+                    if (operationType == OperationType.UPDATE_SHARE_PERMISSIONS ||
+                        operationType == OperationType.UPDATE_PUBLIC_SHARE)
+                        return f.format(R.string.update_link_file_no_exist)
+                    if (operationType == OperationType.CREATE_PUBLIC_SHARE)
+                        return f.forbidden(R.string.share_link_forbidden_permissions)
+                    if (operationType == OperationType.REMOVE_SHARE)
+                        return f.forbidden(R.string.unshare_link_forbidden_permissions)
+                    return if (operationType == OperationType.UPDATE_SHARE_PERMISSIONS ||
+                        operationType == OperationType.UPDATE_PUBLIC_SHARE)
+                        f.forbidden(R.string.update_link_forbidden_permissions)
+                    else f.format(R.string.move_file_invalid_into_descendent)
+                }
+                RemoteOperationResult.ResultCode.SHARE_FORBIDDEN -> {
+                    if (operationType == OperationType.CREATE_PUBLIC_SHARE)
+                        return f.forbidden(R.string.share_link_forbidden_permissions)
+                    if (operationType == OperationType.REMOVE_SHARE)
+                        return f.forbidden(R.string.unshare_link_forbidden_permissions)
+                    return if (
+                        operationType == OperationType.UPDATE_SHARE_PERMISSIONS ||
+                        operationType == OperationType.UPDATE_PUBLIC_SHARE
+                    ) f.forbidden(
+                        R.string.update_link_forbidden_permissions
+                    ) else f.format(R.string.move_file_invalid_into_descendent)
+                }
+                RemoteOperationResult.ResultCode.INVALID_MOVE_INTO_DESCENDANT -> return f.format(R.string.move_file_invalid_into_descendent)
+                RemoteOperationResult.ResultCode.INVALID_OVERWRITE -> {
+                    if (operationType == OperationType.MOVE_FILE) return f.format(R.string.move_file_invalid_overwrite)
+                    return if (operationType == OperationType.COPY_FILE) f.format(R.string.copy_file_invalid_overwrite)
+                    else f.format(R.string.move_file_error)
+                }
+                RemoteOperationResult.ResultCode.CONFLICT -> return f.format(R.string.move_file_error)
+                RemoteOperationResult.ResultCode.INVALID_COPY_INTO_DESCENDANT ->
+                    return f.format(R.string.copy_file_invalid_into_descendent)
+                else -> return getCommonMessageForResult(operationType, resultCode, resultException, resources)
```

Comment:
Apply when instead of so many if else statements in this file

BTW, I think return keyword can be omitted ;-)

Reply:
Done